public/javascript/main.js

```diff
@@ -377,4 +374,99 @@ $(function(){
           $(this).data('chart', chart);
       }
   })
+
+// Tracks interaction with filter field
+// http://stackoverflow.com/questions/4220126/run-javascript-function-when-user-finishes-typing-instead-of-on-key-up
+// Run javascript function when user finishes typing instead of on key up?
+//setup before functions
+//
+var typingTimer;                //timer identifier
+var doneTypingInterval = 2000;  //time in ms, 2 seconds
+var $input = $('[data-ga-track-change]');
+
+//on keyup, start the countdown
+$input.on('keyup', function () {
+  clearTimeout(typingTimer);
+  if (this.value.length > 0)
+    typingTimer = setTimeout(doneTyping, doneTypingInterval);
+});
+
+//on keydown, clear the countdown 
+$input.on('keydown', function () {
+  clearTimeout(typingTimer);
 });
+
+function doneTyping () {
+  ga('send', 'event', 'user input', $(this).attr('data-ga-tack-change'), document.title);
+}
+
+$('[data-ga-track-click]').on('click', deferredTracking)
+
+function deferredTracking(event){
+    event.preventDefault();
+    var href = $(this).attr('href') || false;
+
+    eventDescription = $(this).attr('data-ga-track-click') || 'Undescribed event';
+
+    var deferred = analytics.trackEvent({
+        hitType: 'event',
+        eventCategory: event.type,
+        eventAction: eventDescription,
```

Comment:
Just want to check you’re sure about the order of `eventCategory` and `eventAction` here?

As it stands, we'll be recording an `eventCategory` like `"click"` and an `eventAction` like `"Selection toggle: identifiers"`.

Typically you'd use them the other way round – you’d use `eventAction` to record the action the user took (`"click"`) and `eventCategory` to record the type of object they were interacting with (`"Selection toggle: identifiers"`).